### ISSUE TITLE
[New Concept Exercise] : sequences and common sequence operations

### ISSUE THREAD

**Issue by DavidGerva** [exercism/python#2290]
**Note:  this issue was ported from the working [V3] repo, and needs to be re-done.**

This issue describes how to implement the `slicing` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [Contributing to Exercism](https://exercism.org/docs/building) | [Exercism and GitHub](https://exercism.org/docs/building/github) | [Contributor Pull Request Guide](https://exercism.org/docs/building/github/contributors-pull-request-guide)
- [What are those Weird Task Tags about?](https://exercism.org/docs/building/product/tasks)
- [Building Language Tracks: An Overview](https://exercism.org/docs/building/tracks) 
- [What are Concepts?](https://exercism.org/docs/building/tracks/concepts)
- [Concept Exercise Specifications](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concept Specifications](https://exercism.org/docs/building/tracks/concepts) 
- [Exercism Formatting and Style Guide](https://exercism.org/docs/building/markdown/style-guide)
- [Exercism Markdown Specification](https://exercism.org/docs/building/markdown/markdown)
- [Reputation](https://exercism.org/docs/using/product/reputation)  

## Goal

The goal of this exercise is to teach the student about  the `slicing` operation in Python and how it applies to various data types.

## Learning objectives

*  what is a slice (_a copy of a selected subsection_).
*  understand _slice syntax_ (**e.g.** `[start : stop : step]`), and how Python evaluates slices.
*  understand indexing from the _left_ **and** the _right_, and how this applies to slicing.
* understand which built-in data types & data structures can be sliced, and which _cannot_. (_strings can, numbers cannot_).
* perform slice operations on strings.
* perform a slice operations on _sequence types_ (`lists`, `sets`, & `tuples`).
* copy an object via the "full" slice `[:]`
* reverse an object order via the "reverse slice" `[::-1]`

## Out of scope

* [slice assignment](https://www.learnbyexample.org/python-list-slicing/)
* `slice()` function, and how it differs from `[ ]` notation
* `slice` objects
* `itertools.islice()`

## Concepts

* slicing
* indexing

## Prerequisites

- `basics`
-  `iteration`
- `lists`
- `list-methods`
- `sets`
- `strings`
- `tuples`

## Resources to refer to

- https://railsware.com/blog/python-for-machine-learning-indexing-and-slicing-for-lists-tuples-strings-and-other-sequential-types/

### Hints

- TBD?

### After

- the `slice()` function

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named slicing_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
